lib/CType.ml

```diff
@@ -21,6 +21,7 @@ type base = string
 type t =
   | Base of base
   | Volatile of t
+  | Const of t
```

Comment:
Adding the `Const` constructor impacts many functions of this module, which should strip  the new attribute, as they strip `Volatile` and `Atomic`. Those are `is_ptr`, `is_array`, `is_atomic`, strip_atomic`, `strip_volatile`, `strip_attributes` and `is_ptr_to_atomic`. Here is for instance haw `is_ptr` can be rewritten:
```
let rec is_ptr = function
| Atomic t|Volatile t|Const t -> is_ptr t
| Pointer _ -> true
| _ -> false
```

Obviously the initial code is not 1/ very robust 2/ poorly written initially, (adding a constructor should fire so much changes).
 1.  Underscores can be replaced by completing or-patterns. For instance the last clause of `is_ptr` can be:
```
| Array _|Base _ -> false
```
2. More involved, one can think about having a `Attribute of attribute * t` constructor (with `type attribute = Volatile|Atomic|Const`), and/or restructuring code (_e.g._ the "is_*" functions could call strip_attributes.